caravel/assets/javascripts/components/ModalTrigger.jsx

```diff
@@ -0,0 +1,55 @@
+import React, { PropTypes } from 'react';
+import { Modal } from 'react-bootstrap';
+
+const propTypes = {
+  buttonBody: PropTypes.node.isRequired,
+  modalTitle: PropTypes.string.isRequired,
+  modalBody: PropTypes.node.isRequired,
+  beforeOpen: PropTypes.func,
+};
+
+const defaultProps = {
+  beforeOpen: () => {},
+};
+
+export default class ModalTrigger extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      showModal: false,
+    };
+    this.open = this.open.bind(this);
+    this.close = this.close.bind(this);
+  }
+
+  close() {
+    this.setState({ showModal: false });
+  }
+
+  open() {
+    this.props.beforeOpen();
+    this.setState({ showModal: true });
+  }
+
+  render() {
+    return (
+      <span
```

Comment:
this component is great but on my side I wanted to use it not as a button (just an `<i/>`). Maybe instead of receiving `buttonBody` is could receive `trigger` as any jsx, but still be wrapped in an `<span onClick>`.

Another option is to put `this.props.children` inside the onClick span as opposed to `this.props.trigger`.